In [1]:
from pathlib import Path
from abstract_search_evaluation import get_scores

/home/adamp/school/cmpt419/layman-scientific-embeddings/venv/lib/python3.12/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [2]:
MODELS_PATH = Path("models")

In [3]:
def print_model_scores(model_path: str | Path):
    scores = get_scores(model_path)
    print(model_path)
    print("Full match score:", scores["perfect_match_score"])
    print("Related keywords score:", scores["related_keyword_score"])

In [4]:
for model_path in MODELS_PATH.iterdir():
    print_model_scores(model_path)
    print("")

You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.
Loading weights: 100%|██████████| 199/199 [00:00<00:00, 902.69it/s, Materializing param=pooler.dense.weight]                               
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/vanilla-finetuned
Full match score: 0.05533596837944664
Related keywords score: 0.7924901185770751



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 907.78it/s, Materializing param=pooler.dense.weight]                               
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/jargon-layman-finetuned-0.50
Full match score: 0.05731225296442688
Related keywords score: 0.8221343873517787



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 889.72it/s, Materializing param=pooler.dense.weight]                               
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/jargon-layman-finetuned-0.75
Full match score: 0.06126482213438735
Related keywords score: 0.8330039525691699



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 995.95it/s, Materializing param=pooler.dense.weight]                               
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/jargon-layman-finetuned-1.00
Full match score: 0.06324110671936758
Related keywords score: 0.849802371541502



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 922.86it/s, Materializing param=pooler.dense.weight]                               


models/jargon-layman-finetuned-0.25
Full match score: 0.06324110671936758
Related keywords score: 0.8142292490118577



In [5]:
print_model_scores("allenai/scibert_scivocab_uncased")

No sentence-transformers model found with name allenai/scibert_scivocab_uncased. Creating a new one with mean pooling.
Loading weights: 100%|██████████| 199/199 [00:00<00:00, 1292.41it/s, Materializing param=pooler.dense.weight]                               
BertModel LOAD REPORT from: allenai/scibert_scivocab_uncased
Key                                        | Status     |  | 
-------------------------------------------+------------+--+-
cls.predictions.bias                       | UNEXPECTED |  | 
cls.predictions.decoder.weight             | UNEXPECTED |  | 
cls.seq_relationship.bias                  | UNEXPECTED |  | 
cls.predictions.transform.dense.bias       | UNEXPECTED |  | 
cls.seq_relationship.weight                | UNEXPECTED |  | 
cls.predictions.transform.dense.weight     | UNEXPECTED |  | 
cls.predictions.transform.LayerNorm.bias   | UNEXPECTED |  | 
cls.predictions.decoder.bias               | UNEXPECTED |  | 
cls.predictions.transform.LayerNorm.weight | UNEXPECTED |  

allenai/scibert_scivocab_uncased
Full match score: 0.009881422924901186
Related keywords score: 0.11462450592885376


# Results

| Model | Exact keyword match score | Related keyword score |
|---|---|---|
| allenai/scibert_scivocab_uncased | 0.01 | 0.11 |
| vanilla-finetuned | <u>0.06</u> | **0.79** |
| jargon-layman-finetuned | <u>0.06</u> | <u>0.85</u> |